### ISSUE TITLE
error when opening a file that's directory does not yet exist

### ISSUE THREAD

**Issue by dpassen** [liuyinz/mise.el#1]
When opening a file in a directory that doesn't yet exist, emacs normally will open a buffer to the file and suggest you run `M-x make-directory`, but when `global-mise-mode` is enabled, the buffer is not opened and the following is printed to the `*Messages*` buffer:
```
mise--call: Setting current directory: No such file or directory, <PATH TO MISSING DIRECTORY>
```

**Comment by liuyinz** [comment]
I will handle the problem after a while, but this issue exists in many scenarios, and the following is a general solution:
```elisp
  ;; ;; SEE https://emacsredux.com/blog/2022/06/12/auto-create-missing-directories/
  (defun my/auto-create-missing-dirs ()
    (let ((target-dir (file-name-directory buffer-file-name)))
      (unless (file-exists-p target-dir)
        (make-directory target-dir t))))
  (add-to-list 'find-file-not-found-functions #'my/auto-create-missing-dirs)

```

**Comment by liuyinz** [comment]
solved in f94860f7e3b6fdaea7eb79beff8f4662f137752b, now new buffer without existed directory would not start mise-mode by default, and a warning message would popup.

**Comment by dpassen** [comment]
Thank you!
